src/chi/javascript/components/sidenav.js

```diff
@@ -192,10 +198,27 @@ class Sidenav extends Component {
     if (activator && drawer.id) {
       menuItemLink = this._getMenuItemLink(drawer);
       menuItem = menuItemLink.parentNode;
-      this.close(menuItemLink);
+      if (drawerMenuItem.querySelector(`.${DRAWER_ITEM_LIST_CLASS}`) === null ||
+        e.target.classList.contains('a-sidenav__drawer-item-tab')) {
+        this.close(menuItemLink);
+      }
 
       if (drawerMenuItem) {
```

Comment:
Can the Drawer menu item list function as a toggle? If expanded, clicking again would collapse

Reply:
@janibolkvadze Collapsing effect is very slow. Can we eliminate that delay? We should also probably remove the `-active` class after collapsing.